controllers/competition_supervisor/competition_supervisor.py

```diff
@@ -33,6 +34,11 @@ def record_video(supervisor: Supervisor, file_path: Path) -> Iterator[None]:
     print("Saving video to {}".format(file_path))
 
     config = controller_utils.get_recording_config()
+    if 'SR_COMP_NO_RECORD' in os.environ:
```

Comment:
The alternative of using quality==0, as Peter suggested:
```suggestion
    if config.quality == 0:
```

Reply:
Personally i'd go for reading the quality, or deriving it from a separate key. Adding an argument to `run_comp_match` a la `--no-record` would then be simple to implement and thus easy to set on CI.

Reply:
> Adding an argument to `run_comp_match` a la `--no-record` would then be simple to implement and thus easy to set on CI.

I'm not sure we'd want to set this in CI. Part of the goal of CI was to fully replicate the running of matches to ensure that they weren't broken. Manual inspection of the resulting videos is very much part of that.

If we want to expand the CI running to include other forms of testing (which I hope we do do), then those may not want to have recordings for speed, though they also likely won't be running as competition matches.

I suggest that we not think about this as a CI-first solution -- I don't see running with video recordings as an issue in CI (rather I think it's desirable).

Reply:
7689e435504abd036a250cbe9ac547ca5dc3eb73

Reply:
> I'm not sure we'd want to set this in CI. Part of the goal of CI was to fully replicate the running of matches to ensure that they weren't broken. Manual inspection of the resulting videos is very much part of that.

I disagree with this, currently the length limitation imposed by the running speed prevents testing actions that don't occur at the start of the match, like the wall controller's functionality. Since the animation is still created, manual inspection of the animation provides the same information as can be gained from the video.

> I suggest that we not think about this as a CI-first solution -- I don't see running with video recordings as an issue in CI (rather I think it's desirable).

But that's what it is, the purpose of this is solely for CI and automated testing.

Reply:
For doing things for the purpose of testing the environment I think we're going to want a rather different setup. I'm expecting that we'll have a supervisor which exists entirely for testing and that supervisor would be in control of the environment. This would let us do things like test "has the robot moved to the right place after it drove there", as it would be able to inspect the properties of all the nodes. In that setup, having the competition supervisor in place too is going to get very complicated.

How had you imagined the test setup working if you still want the competition supervisor to be controlling things?

Reply:
You've misunderstood me, when testing functionality other than competition mode we wouldn't be running in comp mode so the the discussion of inhibiting video creation is moot. I was talking specifically about our current testing of a comp match in CI, and the possibility of automatically testing of team's code uploads.

My point was that, since we have the animations and we aren't going to be livestreaming these matches, we can still obtain all the information required for testing without needing the video creation.